re_data_ui/src/components/LineageGraph.tsx

```diff
@@ -1,36 +1,50 @@
 import React, {
-  memo, PropsWithChildren, ReactElement, useState,
+  memo, PropsWithChildren, ReactElement, useEffect, useState,
 } from 'react';
-import VisNetworkReactComponent from 'vis-network-react';
 import { useSearchParams } from 'react-router-dom';
-import { Network, Options, TimelineEvents } from 'vis';
-import { IGraph, VisNetworkEventParams } from '../pages/GraphView';
+import { Options, TimelineEvents } from 'vis';
+import VisNetworkReactComponent from 'vis-network-react';
+import { IGraph, VisNetworkEventParams } from '../partials/Graph';
 
-interface LineageGraphProps {
+export interface LineageGraphProps {
   data: IGraph;
   events: { [event in TimelineEvents | string]: (params: VisNetworkEventParams) => void };
   networkOptions: Options;
   overviewDataLoaded: boolean;
+  showModelDetails: boolean;
 }
 
 const LineageGraph: React.FC<LineageGraphProps> = (
   props: PropsWithChildren<LineageGraphProps>,
 ): ReactElement => {
-  const [network, setNetwork] = useState<Network | null>(null);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [network, setNetwork] = useState<any>(null);
   const [searchParams] = useSearchParams();
   const {
-    data, overviewDataLoaded, networkOptions, events,
+    data, overviewDataLoaded, networkOptions,
+    events, showModelDetails,
   } = props;
   const model = searchParams.get('model');
-  if (model && network) {
-    network.selectNodes([model], true);
-  }
-  const getNetwork = (n: Network) => {
+  // console.log('data: ', data);
+  useEffect(() => {
+    if (network && model && data) {
+      network?.body?.data?.nodes?.clear();
+      network?.body?.data?.nodes?.add(data.nodes);
+      network?.body?.data?.edges?.clear();
+      network?.body?.data?.edges?.add(data.edges);
+    }
```

Comment:
I would prefer you clear switch the order a bit. To clear nodes and edges first before adding the new ones. Looks to me like there's a brief moment where have new nodes and old edges.

Reply:
okay